### PR TITLE
More Monsters Cleanup

### PR DESCRIPTION
**Pull request by raymondtfr**
More cleanup to monster files.

### CODE REVIEW

File: data/monster/Bosses/the_horned_fox.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@
 	<summons maxSummons="6">
 		<summon name="minotaur archer" interval="1000" chance="13" max="2" />
 		<summon name="minotaur guard" interval="1000" chance="13" max="2" />
```
Comment (marksamman) [inline data/monster/Bosses/the_horned_fox.xml]: Looks like there are more instances of "max" in summon tags, any reason why you didn't remove all of them?

File: data/monster/Bosses/man_in_the_cave.xml [hunk 2]
```diff
@@ -40,7 +40,7 @@
 		<voice sentence="A MIC to rule them all!" />
 	</voices>
 	<summons maxSummons="2">
-		<summon name="Monk" interval="2000" chance="20" max="1" />
+		<summon name="Monk" interval="2000" chance="20" />
```
Comment (ranisalt) [inline data/monster/Bosses/man_in_the_cave.xml]: Man in the Cave summons only 1 monk, doesn't?

**Comment by marksamman** [comment]
I would prefer parsing the max value and not allowing the monster to summon more than max instances of the specific monster instead of removing it.

File: data/monster/Bosses/man_in_the_cave.xml (hunk 2 above)
Comment (raymondtfr) [inline data/monster/Bosses/man_in_the_cave.xml]: Right now I'm only doing small cleanups so I'm not paying attention to these details as of now.

File: data/monster/Bosses/the_horned_fox.xml (hunk 1 above)
Comment (raymondtfr) [inline data/monster/Bosses/the_horned_fox.xml]: Not really, I just missed them. I am using basic search functions in sublime text as I don't understand much of regular expressions, so I miss some things. I'll try to find more occurrences of it.

By the way, I will be replacing melee attributes "attack" and "skill" with "min" and "max" in a few minutes.

**Comment by raymondtfr** [comment]
I find that better too. But would we still need of ``maxSummons``?

**Comment by marksamman** [comment]
Yes, maxSummons should continue to function as it currently does.

**Comment by raymondtfr** [comment]
So should I remove the ``removed nonexistent summons attribute "max"`` commit?

**Comment by marksamman** [comment]
Only if we actually implement the max feature, otherwise we might just as well remove it.

**Comment by raymondtfr** [comment]
Hmm, it could take a while if I were to implement it since I am not much experienced in C++.

By the way, I have just replaced melee attributes with min, max as it is more clear.
I have found more things to fix, I'll be adding more commits soon.

**Comment by marksamman** [comment]
Let me know when you feel that this is ready to be merged.

**Comment by raymondtfr** [comment]
I'll do.

It seems to me that it is going to take a while though, it so seems way every time I go through the files.

**Comment by ranisalt** [comment]
What was this chance1 for?

**Comment by raymondtfr** [comment]
Just another alias for "chance".

**Comment by marksamman** [comment]
Thanks!

**Pull request merged**